backend/app/assets/javascripts/spree/backend/reports.js

```diff
@@ -63,27 +63,78 @@ function getData (json, key) {
   })
 }
 
+/**
+ * @param {string} reportId String ID of the report, comes from data-id attr.
+ * @returns {string} API URL
+ */
+function getApiURL (reportId) {
+  return '/admin/reports/' + reportId + '.json' + window.location.search
+}
+
 /**
  * @param {object} data Data referencing information for data fetching
  * @param {Chart} chart Chart instance
  */
 function updateChart (data, chart) {
-  return fetch(`/admin/reports/${data.id}.json`)
+  return fetch(getApiURL(data.id))
     .then(function (response) {
       return response.json()
     })
     .then(function (json) {
       chart.data.datasets[0].data = getData(json, data.dataKey)
       chart.data.labels = getData(json, data.labelKey)
-    }).finally(function () {
+    })
+    .finally(function () {
       chart.update()
     })
 }
 
+/**
+ * @param {HTMLSelectElement} node DOM select not to get index of
+ * @param {string} param The param from URL to search through select options
+ * @returns {number} Index of option value that should be selected
+ */
+function getParamSelectedIndex (node, param) {
+  const options = []
+  const selectedValue = new Uri(window.location.search).getQueryParamValue(param)
+
+  Array.prototype.forEach.call(node.options, function (el) {
```

Comment:
Maybe we could use `map` method here?

Reply:
@piotrleniec-spark We can't use `map` over `HTMLOptionsCollection`. We have to turn it into `Array` first. This is ES5 way of simplifying this.